### DISCUSSION TITLE
Fixed length strings

### DISCUSSION THREAD

**Discussion by ischoegl** [BlueBrain/HighFive#730] [Questions] [answered]
I have some HDF files that were written by `h5py` that contain fixed length strings, e.g.

```
$ h5dump somefile.h5
[...]
      DATASET "spam" {
         DATATYPE  H5T_STRING {
            STRSIZE 4;
            STRPAD H5T_STR_NULLPAD;
            CSET H5T_CSET_ASCII;
            CTYPE H5T_C_S1;
         }
         DATASPACE  SIMPLE { ( 3 ) / ( 3 ) }
         DATA {
         (0): "eggs", "eggs", "eggs"
         }
      }
[...]
```

This could presumably be read with something along the lines of `HighFive::FixedLenStringArray<4> v;`, which assumes a priori knowledge of the string length to use the template. However, I don't know that length as it is flexible and not known at compile time. I am probably overlooking some other option?

**Comment by 1uc** [comment]
I'm not sure about strings, we don't do them very well, so I'm not very optimistic. Maybe @alkino knows?

**Comment by alkino** [comment]
There is already this discussion there, and it is not working.
You can check the size before and then read.
https://github.com/BlueBrain/HighFive/issues/505

**Comment by 1uc** [comment, marked as answer]
With https://github.com/BlueBrain/HighFive/pull/744 merged std::strings and containers thereof, e.g. std::vector<std::string> can be read and written to fixed or variable length HDF5 strings. There's an example here:
https://github.com/BlueBrain/HighFive/blob/master/src/examples/read_write_std_strings.cpp
